interfaces/builtin/optical_drive.go

```diff
@@ -28,27 +37,73 @@ const opticalDriveBaseDeclarationSlots = `
         - core
 `
 
-const opticalDriveConnectedPlugAppArmor = `
-# Allow read access to optical drives
-/dev/sr[0-9]* r,
-/dev/scd[0-9]* r,
-@{PROC}/sys/dev/cdrom/info r,
-/run/udev/data/b11:[0-9]* r,
-`
+// opticalDriveInterface is the type for optical drive interfaces.
+type opticalDriveInterface struct {
+	commonInterface
+}
+
+// BeforePrepareSlot checks and possibly modifies a slot.
+// Valid "optical-drive" slots may contain the attribute "write".
+// If defined, the attribute "write" must be either "true" or "false".
+func (iface *opticalDriveInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	write, ok := plug.Attrs["write"].(bool)
+	if !ok {
+		if plug.Attrs == nil {
+			plug.Attrs = make(map[string]interface{})
+		}
+		write = false
+		plug.Attrs["write"] = false
+	}
+
+	if write != true && write != false {
+		return fmt.Errorf("optical-drive write attribute must be either undefined, true, or false.")
+	}
+
+	return nil
```

Comment:
In other interfaces we raise an error rather than setting the Attrs to a default value. I suggest something along the lines of:
```
func (iface *opticalDriveInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
       // It's fine if 'write' isn't specified, but if it is, it needs to be bool
       if w, ok := plug.Attrs["write"]; ok {
               _, ok = w.(bool)
               if !ok {
                       return fmt.Errorf(`optical-drive plug requires "write" be bool`)
               }
       }

       return nil
}
```